### PR TITLE
Define tags

### PR DESCRIPTION
**Pull request by nahratzah**
Allows defining tags during a define statement.

Example:

    define my.group tag(instance = sample.metric app_name) {
        cpu = sample.metric cpu;
        mem = sample.metric mem;
    }

### CODE REVIEW

File: impl/src/main/java/com/groupon/lex/metrics/config/DerivedMetricStatement.java [hunk 1]
```diff
@@ -66,6 +68,12 @@ public StringBuilder configString() {
                 .append(impl_.getGroup().configString())
                 .append(' ');
 
+        if (!impl_.getTags().isEmpty()) {
+            sb.append(impl_.getTags().entrySet().stream()
+                    .map(tag_mapping -> INDENT + ConfigSupport.maybeQuoteIdentifier(tag_mapping.getKey()) + " = " + tag_mapping.getValue().configString())
```
Comment (dmjb) [inline impl/src/main/java/com/groupon/lex/metrics/config/DerivedMetricStatement.java]: Define `ConfigSupport.maybeQuoteIdentifier(tag_mapping.getKey())` and `tag_mapping.getValue().configString()` as variables inside the if block to improve readability.

Comment (nahratzah) [inline impl/src/main/java/com/groupon/lex/metrics/config/DerivedMetricStatement.java]: Done.

I made the corresponding change to the similar code that writes metrics too.

File: impl/src/main/java/com/groupon/lex/metrics/config/impl/DerivedMetricTransformerImpl.java [hunk 2]
```diff
@@ -34,64 +34,91 @@
...
-                            .map(path -> MetricName.valueOf(path.getPath()))
-                            .map(name -> SimpleMapEntry.create(name, name_expr.getValue()));
-                })
-                .flatMap(opt -> opt.map(Stream::of).orElseGet(Stream::empty))
+
+        Map<String, MetricValue> scalar_tags = new HashMap<>();
+        Map<Tags, Map<String, MetricValue>> vector_tags = new HashMap<>();
+        resolveMapping(ctx, tags, Optional::of)
+                .forEach((Map.Entry<String, TimeSeriesMetricDeltaSet> tag) -> {
+                    tag.getValue().asScalar()
```
Comment (dmjb) [inline impl/src/main/java/com/groupon/lex/metrics/config/impl/DerivedMetricTransformerImpl.java]: move the body of this lambda into a private static method.

Comment (nahratzah) [inline impl/src/main/java/com/groupon/lex/metrics/config/impl/DerivedMetricTransformerImpl.java]: I moved most of the Tagging logic to an object instead, hope you like it. :)

**Pull request merged**